eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/AggregateBasedStorageEngineTestSuite.java

```diff
@@ -219,16 +224,94 @@ void eventsWithTagsNotMatchingCriteriaAreInsertedAtSequenceZero() {
     }
 
     @Test
-    void transactionRejectedWithConflictingEventsInStore() {
-        testSubject.appendEvents(AppendCondition.withCriteria(TEST_AGGREGATE_CRITERIA),
+    void sourcingFromTwoAggregateStreamsReturnsACombinedStream() {
+        var marker = testSubject.appendEvents(AppendCondition.none(),
+                                              taggedEventMessage("event-0", TEST_AGGREGATE_CRITERIA.tags()),
+                                              taggedEventMessage("event-1", OTHER_AGGREGATE_CRITERIA.tags()),
+                                              taggedEventMessage("event-2", TEST_AGGREGATE_CRITERIA.tags()),
+                                              taggedEventMessage("event-3", OTHER_AGGREGATE_CRITERIA.tags()),
+                                              taggedEventMessage("event-4", TEST_AGGREGATE_CRITERIA.tags()),
+                                              taggedEventMessage("event-5", OTHER_AGGREGATE_CRITERIA.tags()))
+                                .thenCompose(AppendTransaction::commit).join();
+
+        Set<String> actual = new HashSet<>();
+        AtomicReference<ConsistencyMarker> lastMarker = new AtomicReference<>();
+
+        StepVerifier.create(testSubject.source(SourcingCondition.conditionFor(TEST_AGGREGATE_CRITERIA,
+                                                                              OTHER_AGGREGATE_CRITERIA)).asFlux())
+                    //there is no predefined order between aggregates. We just expect 6 entries.
```

Comment:
Curious: why move the assertion out of the `StepVerifier`?

Reply:
Aaah, so prep-work on something that's not yet in place. Since, as far as I know, we do not merge two message streams completely for legacy stores. Instead, we first do one call, then to the other, leading to an append.
Nonetheless, it makes sense to already adjust the test to be resilient against that future change.